app/reconciler/readRuote53IntoCompareStructure.server.ts

```diff
@@ -0,0 +1,70 @@
+import { set } from 'lodash';
+import { getRecordPage } from '~/lib/dns.server';
+
+// Using this in JS code later, cannot `import type`
+import { DnsRecordType } from '@prisma/client';
+import type { ReconcilerCompareStructure } from './ReconcilerTypes';
+import type { ResourceRecordSet, ListResourceRecordSetsResponse } from '@aws-sdk/client-route-53';
+
+/**
+ * !!! There is cross-function object mutation in this file.
+ * This is a huge anti-pattern, but in this case it made the code
+ * much more performant. So I marked the variable as such and
+ * doing the mutation either way
+ */
+
+const processRecordSets = (
+  recordSets: ResourceRecordSet[],
+  // !!! `MUTATEDcompareStructure` is passed by reference and is mutated !!!
+  MUTATEDcompareStructure: ReconcilerCompareStructure
+) => {
+  recordSets.forEach((recordSet) => {
+    // Unsure as to how those could be undefined, but according to AWS sdk, they could
+    if (!recordSet.Name || !recordSet.Type) {
+      return;
+    }
+
+    // We only care about record types that we handle. NS, CNAME, etc. should be ignored
```

Comment:
We include `CNAME`, why have you listed it here with `NS`?

Reply:
Typo, I meant SOA